metadata-editor/app/lib/Syndication.scala

```diff
@@ -0,0 +1,148 @@
+      photoshootMaybe <- getPhotoshootForImage(id)
+      imagesInPhotoshoot <- getImagesInPhotoshoot(photoshootMaybe)
+      _ <- syndicationStore.jsonAdd(id, "usageRights", DynamoDB.caseClassToMap(syndicationRight))
+      _ = publish(imagesInPhotoshoot, UpdateImageSyndication)
+    } yield syndicationRight
+
+  def getSyndicationForImage(id: String)
+                            (implicit ec: ExecutionContext): Future[Option[SyndicationRights]] = {
+    syndicationStore.jsonGet(id, syndicationRights)
+      // It's OK for the image to _not_ exist in the syndication store, so this needs to be recovered
+      .recover { case NoItemFound => JsNull }
+      .flatMap(dynamoEntry => (dynamoEntry \ syndicationRights).toOption match {
+        // If image has its own rights, return those rights, with isInferred false
+        case Some(rights) => Future.successful(Some(rights.as[SyndicationRights].copy(isInferred = false)))
+        // If the image does not have it's own rights, get rights for the photoshoot
+        case None => getRightsByPhotoshoot(id)
+      })
+  }
+
+  def getRightsByPhotoshoot(id: String)
+                           (implicit ec: ExecutionContext): Future[Option[SyndicationRights]] = getPhotoshootForImage(id)
+    // It's ok for the image to _not_ exist in the edits store - it may have no photoshoot (or any other edit)
+    .recover{ case NoItemFound => None }
+    .flatMap {
+      //  If image is not in a photoshoot, return no rights
+      case None => Future.successful(None)
+      //  If image is in a photo shoot, find the most recently published image and return those rights, with isInferred true
+      case Some(photoshoot) => getMostRecentInferredSyndicationRights(photoshoot)
+    }
+
+  private def getMostRecentInferredSyndicationRights(photoshoot: Photoshoot)
+                                                    (implicit ec: ExecutionContext): Future[Option[SyndicationRights]] = {
+    getImagesInPhotoshoot(Some(photoshoot)) flatMap {
+      ids =>
+        getRightsForImages(ids)
+          .map(list => getMostRecentSyndicationRights(list))
+          .map(rightsMaybe => rightsMaybe.map(rights => rights.copy(isInferred = true)))
+    }
+  }
+
+  private def getRightsForImages(ids: List[String])
+                                (implicit ec: ExecutionContext) = {
+    Future.traverse(ids)(id => syndicationStore.jsonGet(id, syndicationRights)
+      // Any/all of the images in this photoshoot may have no rights, so recover
+      .recover { case NoItemFound => JsNull }
+      .map(dynamoEntry =>
+        (dynamoEntry \ syndicationRights).toOption map (x => x.as[SyndicationRights]))
+    )
+  }
+
+  def getMostRecentSyndicationRights(list: List[Option[SyndicationRights]]): Option[SyndicationRights] = list
+    .collect {case Some(sr) => sr}.filter(_.published.nonEmpty).sortBy(_.published.map(-_.getMillis)).headOption
+
+  def filterImagesByInference(ids: List[String], inferred: Boolean)
```

Comment:
I'm not quite sure what this is doing. I think we might be caching inferred rights in the syndication store and this is a mechanism for getting stored actual rights or cached inferred rights? If so then I'd challenge the idea that we cache anything as that's a potentially lovely source of bugs. I'd prefer to only store actual rights and then we can do the more expensive calculation as required.

If we want to be performant we could only publish the ID of the directly modified image and then return the complete calculation of all touched images so thrall can do a batch update?